### PR TITLE
188 redo adding owner item to dataset and delete by only owner

### PR DESCRIPTION
**Pull request by ywkim312**
To test this PR:

1. Create any dataset, hazard, fragility curves, or anything. 
2. After the new dataset created, check if there is owner part in the metadata. It should be there and the owner should be your name
3. Try to delete the newly created dataset. However, test it with the different username. You can change the preffered_username in the header to someone else or just not the owner.
4. It should fail, if the preffered_username is not the same as the owner.
5. Use the owner as preffered_username, then it should be deleted

### CODE REVIEW

**Review by ylyangtw — APPROVED** [review]
Changes look good. Endpoints POST/DELETE are working. Author info is correct. Approve!

File: server/incore-common/src/main/java/edu/illinois/ncsa/incore/common/utils/UserGroupUtils.java [hunk 1]
```diff
@@ -30,4 +30,13 @@ public static List<String> getUserGroups(String userGroups) {
             throw new IncoreHTTPException(Response.Status.BAD_REQUEST, "Could not map provided user-group.");
         }
     }
+
+    public static boolean isAdmin(List<String> groups) {
```
Comment (longshuicy) [inline server/incore-common/src/main/java/edu/illinois/ncsa/incore/common/utils/UserGroupUtils.java:34]: I think we already have a function for checking admin in `Authorizer.java`
```
 public boolean isUserAdmin(List<String> userGroups) {
        return (userGroups.contains("incore_admin") || userGroups.contains("incore_ncsa"));
    }
```

Comment (ywkim312) [reply server/incore-common/src/main/java/edu/illinois/ncsa/incore/common/utils/UserGroupUtils.java]: change all isAdmin boolean to use isUserAdmin in Authorizer.java

**Review by longshuicy — APPROVED** [review]
I tested the below scenario and it works:
1. create a DFR3 mapping using non-admin account "commresilience"
2. went into the data base and change the owner to something else, say "incore"
3. delete this mapping fail because now owner is not the same
4. login with admin acount and successfully delete the mapping (since admin can bypass)

Code looks good as well. Approve.

**Pull request merged**